pdata/pprofile/string_table.go

```diff
@@ -0,0 +1,33 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package pprofile // import "go.opentelemetry.io/collector/pdata/pprofile"
+
+import (
+	"errors"
+	"math"
+
+	"go.opentelemetry.io/collector/pdata/pcommon"
+)
+
+var errTooManyStringTableEntries = errors.New("too many entries in StringTable")
+
+// SetString updates a StringTable, adding or providing a value and returns its index.
+func SetString(table pcommon.StringSlice, val string) (int32, error) {
```

Comment:
LoadOrStore to be consistent with https://pkg.go.dev/sync#Map.LoadOrStore?

Reply:
Every Set checks whether the value exists or not, to avoid duplicates.
The difference here is that we can't set the indices value, since many different field names can store strings.

So we set the value in the StringSlice and return the index, to be set by the caller.

Reply:
I would suggest we got with `Set` given the precedent and we create an issue to make a decision on this in the future. Since `pprofile` is still 0.x we can always change it later